lib/Widgets/Settings.vala

```diff
@@ -96,9 +115,46 @@ namespace Granite {
             }
         }
 
+        private void setup_accent_color () {
+            try {
+                if (portal == null) {
+                    portal = Portal.Settings.get ();
+                }
+
+                var variant = portal.read (
+                    "org.freedesktop.appearance",
+                    "accent-color"
+                ).get_variant ();
+
+                accent_color = parse_color (variant);
+
+                portal.setting_changed.connect ((scheme, key, value) => {
+                    if (scheme == "org.freedesktop.appearance" && key == "accent-color") {
+                        accent_color = parse_color (value);
+                    }
+                });
+            } catch (Error e) {
+                warning (e.message);
+
+                // Set a default in case we can't get from system
+                accent_color.parse ("#3689e6");
```

Comment:
Doesn't we need `accent_color = Gdk.RGBA ();` here? I guess, otherwise, `accent_color` would remain null and this line could cause null access when `_accent_color` is null and `setup_accent_color ()` is called.